File: apprenticeship/apprenticeship/report/trainee_date_of_joining_or_leaving/trainee_date_of_joining_or_leaving.py
from __future__ import unicode_literals
from . import create_or_update_report

def get_report_data():
    return {
        "name": "Trainee Date of Joining or Leaving",
        "ref_doctype": "Trainee",
        "report_type": "Query Report",
        "query": """SELECT
                name as "Trainee:Link/Trainee:150",
                full_name as "Trainee Name:Data:180",
                date_of_engagement as "Date of Engagement:Date:130",
                date_of_resignation as "Date of Resignation:Date:130",
                dealer as "Dealer:Link/Dealer:150"
            FROM `tabTrainee`
            ORDER BY date_of_engagement DESC""",
        "roles": [{"role": "Education Manager"}, {"role": "Instructor"}, {"role": "System Manager"}],
        "add_total_row": 0,
        "disable_prepared_report": 0,
        "prepared_report": 0,
    }

def create():
    create_or_update_report(get_report_data())

File: apprenticeship/apprenticeship/report/trainee_date_of_joining_or_leaving/trainee_date_of_joining_or_leaving.js
frappe.query_reports["Trainee Date of Joining or Leaving"] = {
	"filters": [
		{
			"fieldname": "dealer",
			"label": __("Dealer"),
			"fieldtype": "Link",
			"options": "Dealer"
		},
		{
			"fieldname": "engagement_from",
			"label": __("Engagement From"),
			"fieldtype": "Date"
		},
		{
			"fieldname": "engagement_to",
			"label": __("Engagement To"),
			"fieldtype": "Date"
		},
		{
			"fieldname": "resignation_from",
			"label": __("Resignation From"),
			"fieldtype": "Date"
		},
		{
			"fieldname": "resignation_to",
			"label": __("Resignation To"),
			"fieldtype": "Date"
		}
	]
};